pypdf/_page.py

```diff
@@ -1868,6 +1878,83 @@ def process_operation(operator: bytes, operands: List[Any]) -> None:
             visitor_text(text, memo_cm, memo_tm, cmap[3], font_size)
         return output
 
+    def _layout_mode_fonts(self) -> Dict[str, _layout_mode.Font]:
+        """
+        Get fonts formatted for "layout" mode text extraction
+
+        Returns:
+            Dict[str, Font]: dictionary of _layout_mode.Font instances keyed by font name
+        """
+        # Font retrieval logic adapted from pypdf.PageObject._extract_text()
+        objr: Any = self
+        while NameObject(PG.RESOURCES) not in objr:
+            objr = objr["/Parent"].get_object()
+        resources_dict: Any = objr[PG.RESOURCES]
+        fonts: Dict[str, _layout_mode.Font] = {}
+        if "/Font" in resources_dict and self.pdf is not None:
+            for font_name in resources_dict["/Font"]:
+                *cmap, font_dict_obj = build_char_map(font_name, 200.0, self)
+                font_dict = {
+                    k: self.pdf.get_object(v)
+                    if isinstance(v, IndirectObject)
+                    else [
+                        self.pdf.get_object(_v) if isinstance(_v, IndirectObject) else _v
+                        for _v in v
+                    ]
+                    if isinstance(v, ArrayObject)
+                    else v
+                    for k, v in font_dict_obj.items()
+                }
+                # mypy really sucks at unpacking
+                fonts[font_name] = _layout_mode.Font(*cmap, font_dict)  # type: ignore
```

Comment:
Is there anything we can do here about not ignoring the typing stuff completely?

Reply:
I've narrowed the ignore directive.

The types returned by the call to build_char_map() where cmap is defined (just above the font_dict comprehension) *are explicit*. I believe that mypy should support this *extremely common* use case, but it's notorious for its failure to do so (as it is for its failure to properly interpret `zip` and `map` operations). Pylance is able to collect the proper type hints for cmap's contents. Why can't mypy?

Long story short, I'm generally against using an "uglier" code style just to paper over shortcomings in the mypy engine, so I'd prefer to leave as is.  Thoughts?